sycl/test-e2e/Basic/built-ins/helpers.hpp

```diff
@@ -34,14 +34,14 @@ void test(bool CheckDevice, double delta, FuncTy F, ExpectedTy Expected,
   sycl::buffer<bool, 1> SuccessBuf{1};
 
   // Make sure we don't use fp64 on devices that don't support it.
-  sycl::detail::get_elem_type_t<ExpectedTy> d(delta);
-
-  sycl::queue{}.submit([&](sycl::handler &cgh) {
+  sycl::context ctx;
+  sycl::queue q{ctx, ctx.get_devices()[0]};
+  q.submit([&](sycl::handler &cgh) {
```

Comment:
Why do we need this?

Reply:
> Why do we need this?

Without it, the queue uses the default context as described in the default context extension: https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_default_context.asciidoc.
Core SYCL tests should not rely on extensions so I've explicitly created a context and then created a queue with that context.

Reply:
So what? Lots of tests just do `sycl::queue q;` and it's perfectly legal in SYCL. Are you saying that because of this extension we must rewrite *all* these tests?

Reply:
> So what? Lots of tests just do `sycl::queue q;` and it's perfectly legal in SYCL. Are you saying that because of this extension we must rewrite _all_ these tests?

You're right, the reporter of the tracker was under the impression that the bug was due to the extension so that's why they suggested we rewrite it. I've changed it to use the idiomatic `sycl::queue q;`